cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/ParameterTypeRegistry.java

```diff
@@ -19,7 +19,7 @@
     // Pattern.compile(...).pattern() is not necessary, but it helps us take advantage of the IntelliJ's regexp validation,
     // which detects unneeded escapes.
     private static final List<String> INTEGER_REGEXPS = asList(Pattern.compile("-?\\d+").pattern(), Pattern.compile("\\d+").pattern());
-    private static final List<String> FLOAT_REGEXPS = singletonList(Pattern.compile("-?\\d*(?:[.,]\\d+)?").pattern());
+    private static final List<String> FLOAT_REGEXPS = singletonList(Pattern.compile("[+-]?(?:[0-9]*[.,])?[0-9]+").pattern());
```

Comment:
@aslakhellesoy why did you change this? And why only for Java?

Reply:
When I fiddled with the test I removed I discovered that an undefined step def with `-` would suggest {float}. That's why I changed it. In retrospect I realise that should be done in a different PR, so I'll remove it from this one.